lib/nodejs/esm-utils.js

```diff
@@ -39,48 +39,34 @@ exports.requireOrImport = async (file, esmDecorator) => {
     return formattedImport(file, esmDecorator);
   }
   try {
-    return dealWithExports(await formattedImport(file, esmDecorator));
-  } catch (err) {
+    return require(file);
+  } catch (requireErr) {
     if (
-      err.code === 'ERR_MODULE_NOT_FOUND' ||
-      err.code === 'ERR_UNKNOWN_FILE_EXTENSION' ||
-      err.code === 'ERR_UNSUPPORTED_DIR_IMPORT'
+      requireErr.code === 'ERR_REQUIRE_ESM' ||  requireErr.code === 'ERR_INTERNAL_ASSERTION' ||
+      (requireErr instanceof SyntaxError &&
+        requireErr
+          .toString()
+          .includes('Cannot use import statement outside a module'))
```

Comment:
[Question] Why not go off of `requireErr.code` the way the other checks do?

If there is some reason why it should go off other properties in the error, I'd think checking just `requireErr.message` would be preferable to stringifying the whole thing. In theory someone could name a file `test/Cannot use import statement outside a module.just-to-be-annoying.spec.js`.